aiida/engine/processes/process.py

```diff
@@ -499,37 +526,33 @@ def build_process_type(cls):
 
         return process_type
 
-    def report(self, msg, *args, **kwargs):
+    def report(self, msg: str, *args, **kwargs) -> None:
         """Log a message to the logger, which should get saved to the database through the attached DbLogHandler.
 
         The pk, class name and function name of the caller are prepended to the given message
 
         :param msg: message to log
-        :type msg: str
-
         :param args: args to pass to the log call
-        :type args: list
-
         :param kwargs: kwargs to pass to the log call
-        :type kwargs: dict
+
         """
         message = f'[{self.node.pk}|{self.__class__.__name__}|{inspect.stack()[1][3]}]: {msg}'
         self.logger.log(LOG_LEVEL_REPORT, message, *args, **kwargs)
 
-    def _create_and_setup_db_record(self):
+    def _create_and_setup_db_record(self) -> Union[int, UUID]:
         """
         Create and setup the database record for this process
 
-        :return: the uuid of the process
-        :rtype: :class:`!uuid.UUID`
+        :return: the uuid or pk of the process
+
         """
         self._node = self.get_or_create_db_record()
         self._setup_db_record()
         if self.metadata.store_provenance:
             try:
                 self.node.store_all()
                 if self.node.is_finished_ok:
-                    self._state = ProcessState.FINISHED
+                    self._state = Finished(self, None, True)  # TODO check correct
```

Comment:
This fix needs checking

Reply:
seems correct that it was the wrong type, but not sure if constructing a state like this is correct. @muhrin should this work as intended?

Reply:
That's not for this PR, but should aiida-core be tampering with this`_state` attribute and the process state classes directly?
This seems like an example where plumpy might want to offer an API that avoids this (it looks like `_state` is used only used in 3 places in aiida-core) and decouple the implementation a bit.

Reply:
yeh, its a little misleading that `Process.state` returns the label and not the actual state 😬 but probably "too late" to change that

Reply:
Well, if I understand correctly, then the actual process state class is only used in three places in aiida-core.
If one would delegate any operation on this class to plumpy, then the confusion about what type of process state one needs to deal with in aiida-core would go away (it would always just be dealing with labels).

Reply:
Yes, setting to `ProcessState.FINISHED` was not right.  The general logic to `_state` was that, I really wanted to push clients (even subclasses) to only use `transition_to` for state transitions as this way we could try and guarantee that state transition hooks would be called (e.g. `on_exiting/entering`) which are legtimate places for clients to put code that they always expect to be hit when transitionining. _However_, there need to be exceptions when deserialising a process (e.g in `load_instance_state`) because if it was serialised in that state then the transition already happened at some point and it's OK to just set the `_state` directly.

Sorry for the sloppiness in `this.state` actually returning the state label, this was just me trying to be economical and avoid having to call it `state_label` but I recognise it's confusing when working on the internals of a `Process`.

Anyway, if I'm understanding this code segment it's OK to set the state like this because if the node says it's done, presumably at some point the correct state transition did happen.  That said, I'm not sure how this code state could be reached in the first place (and perhaps it never was if this didn't cause issues before).

@ltalirz , yes, no problem to have a `_set_state()`  method so long as there is a _BIG_ warning to never call it except for the situations I mentioned above.